File: 프로그래머스/unrated/181890. 왼쪽 오른쪽/왼쪽 오른쪽.java
import java.util.Arrays;

class Solution {
    public String[] solution(String[] str_list) {
        int idx = 0;
		boolean isL = true;

		for (int i = 0; i < str_list.length; i++) {
			if (str_list[i].equals("l")) {
				idx = i;
				isL = true;
				break;
			} else if (str_list[i].equals("r")) {
				idx = i;
				isL = false;
				break;
			} else continue;
		}

		return isL ? Arrays.copyOfRange(str_list, 0, idx) : Arrays.copyOfRange(str_list, idx + 1, str_list.length);
    }
}

File: 프로그래머스/unrated/181890. 왼쪽 오른쪽/왼쪽 오른쪽.kt
class Solution {
    fun solution(str_list: Array<String>): Array<String> {
        if (!str_list.contains("r") && !str_list.contains("l")) return emptyArray()

        val l_idx : Int = (str_list.indices).firstOrNull { i:Int -> str_list[i].equals("l") } ?: Int.MAX_VALUE
        val r_idx : Int = (str_list.indices).firstOrNull { i:Int -> str_list[i].equals("r") } ?: Int.MAX_VALUE

        return if (l_idx < r_idx) str_list.sliceArray(0 .. l_idx - 1) else str_list.sliceArray(r_idx + 1 .. str_list.lastIndex)
    }
}